pkgs/top-level/python-packages.nix

```diff
@@ -7856,4 +7856,13 @@ in {
 
 });
 
-in fix' (extends overrides packages)
+# Require normalized attribute names as these are predictable.
+# https://www.python.org/dev/peps/pep-0503/#normalized-names
+#isNormalised = name: (builtins.match "^([A-Z0-9]|[A-Z0-9][A-Z0-9]*[A-Z0-9])$") name == null;
+isNormalised = name: name == requiredName name;
```

Comment:
This is too strict. We now cannot have variants such as `pyqt5_with_qtwebkit` or `pytest_6`. Having these versions at the end is possibly to add with a regex, but variants such as `pyqt5_with_qtwebkit` are not possible to distinguish from non-normalized names.

Reply:
Note evaluation is lazy, so you only get the throw when explicitly requesting a `pytest_6`. Still, it is a problem.

Reply:
One could partially work-around this issue by requiring overriding the Python package set to use one of these variants. At least that works for applications. And libraries should not be able to use multiple variants anyway. The limitation is test dependencies, there you may be able to use different versions.

Reply:
@jonringer there is now a `toVariant` function for declaring when a derivation is a variant that can have a `_suffix`.